x-pack/test/saml_api_integration/apis/security/saml_login.ts

```diff
@@ -600,6 +600,19 @@ export default function({ getService }: FtrProviderContext) {
           .set('Cookie', secondNewCookie.cookieString())
           .expect(200);
       });
+
+      it('should refresh access token even if multiple concurrent requests try to refresh it', async () => {
+        // Send 5 concurrent requests with a cookie that contains an expired access token.
+        await Promise.all(
+          Array.from({ length: 5 }).map((value, index) =>
```

Comment:
**question** is there any value to testing that the token we get is the same for all 5 of these requests?

Reply:
I'm not sure, technically it's internal implementation detail of Elasticsearch/Kibana and for consumer it doesn't really matter whether we get the same or different tokens as long as they are valid. But even if think it's worth it, we'll need to add code that decrypts cookie in tests and extracts token from it relying on internal cookie structure. Or you meant something else?

Reply:
> I'm not sure, technically it's internal implementation detail of Elasticsearch/Kibana and for consumer it doesn't really matter whether we get the same or different tokens as long as they are valid. 

I guess my concern was making sure they were all valid. If we happen to get different tokens back in different requests, can we be sure that they'll all work correctly? You're right that it's an implementation detail...I was considering taking a defensive approach here, given all of the session timeout issues we've had to triage lately.

At the end of the day, it's not a huge deal to me one way or another. I'm fine to leave these as-is if it's too complex to decrypt and check, or if you don't think it's worthwhile to do.